src/wp-content/themes/twentynineteen/sass/blocks/_blocks.scss

```
@@ -950,6 +950,11 @@
 		}
 	}
 
+	//! wp-block-social-links
+	.wp-block-social-links .wp-block-social-link.wp-social-link .wp-block-social-link-anchor {
```

This creates a high specificity with six classes:
`.entry .entry-content .wp-block-social-links .wp-block-social-link.wp-social-link .wp-block-social-link-anchor`

An element selector (combined with `.entry .entry-content`) would be enough to override the underline in the post content:

```suggestion
	a:where(.wp-block-social-link-anchor) {
```